components/Map/index.js

```diff
@@ -0,0 +1,85 @@
+import { MapContainer, Marker, Popup, TileLayer } from "react-leaflet";
+import styled from "styled-components";
+import * as L from "leaflet";
+import "leaflet/dist/leaflet.css";
+import "leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.css";
+import "leaflet-defaulticon-compatibility";
+import CurrentOutivityMarker from "./CurrentOutivityMarker";
+import Link from "next/link";
+import useLocalStorage from "use-local-storage";
+
+export default function Map({ currentOutivity = {}, outivities }) {
+  const [userPosition, setUserPosition] = useLocalStorage("userPosition", {
+    lat: "",
+    lng: "",
+  });
+
+  return (
+    <StyledMapContainer
+      center={userPosition ? [userPosition.lat, userPosition.lng] : ["", ""]}
+      zoom={10}
+      scrollWheelZoom
+    >
+      <TileLayer url="https://tiles.stadiamaps.com/tiles/alidade_smooth/{z}/{x}/{y}{r}.png" />
+
+      {outivities.map((outivity) =>
+        outivity.id === currentOutivity.id ? (
+          <CurrentOutivityMarker
+            key={outivity.id}
+            outivity={outivity}
+            position={[outivity.lat, outivity.long]}
```

Comment:
und die handler-function hier auch mitgeben:
```suggestion
            position={[outivity.lat, outivity.long]}
            onSetUserPosition={handleSetUserPosition}
```